src/app/application.cpp

```diff
@@ -144,13 +144,15 @@ Application::Application(const QString &id, int &argc, char **argv)
     QPixmapCache::setCacheLimit(PIXMAP_CACHE_SIZE);
 #endif
 
+    m_portableModeEnabled = QDir(QCoreApplication::applicationDirPath()).exists(DEFAULT_PORTABLE_MODE_PROFILE_DIR);
+
     validateCommandLineParameters();
 
-    const QString profileDir = m_commandLineArgs.portableMode
+    const QString profileDir = m_portableModeEnabled && m_commandLineArgs.profileDir.isEmpty()
```

Comment:
Also here we seems to need `&& m_commandLineArgs.profileDir.isEmpty()` because else `profile` folder will be used instead of `m_commandLineArgs.profileDir`.